### DISCUSSION TITLE
DFU without mcumgr and SMP server

### DISCUSSION THREAD

**Discussion by hslourenc** [zephyrproject-rtos/zephyr#53762] [Q&A]
Hi!

I'm using an nRF52840 with Zephyr and I would like to know if it's possible to perform a DFU without using mcumgr. I would like to write the update to another memory slot and use MCUboot to perform the update. The reason for this is that I'm trying to update the nRF firmware from an ESP32 with ESP-IDF and FreeRTOS, which don't have an mcumgr client available. I've already included MCUboot and written/read random data to/from another memory slot using flash_write and flash_read. The problem now is what to write to the other memory slot and how to make MCUboot perform the update. Is it the content of zephyr.hex of the update that I should write to the other memory slot? How can I make MCUboot perform the update after I've written the update?

My prj.conf:

```
CONFIG_BOOTLOADER_MCUBOOT=y
CONFIG_FLASH=y
CONFIG_FLASH_MAP=y
```

I'm using nRF Connect 2.2.0 and Zephyr 3.1.99. I'm open to other solutions, as long as it uses ESP32 to update the nRF52840.

**Comment by rftafas** [comment]
Hi!

AFAIK, mcumgr is used when there is the server/client. Also, there is the update hub.

You could create something similar, but instead of getting data from IP stack, it would have to come from the ESP32 (uart or SPI, I assume).

PTAL: https://docs.zephyrproject.org/latest/samples/subsys/mgmt/updatehub/README.html
